kv/dist_sender.go

```diff
@@ -731,7 +731,11 @@ func (ds *DistSender) Send(_ context.Context, call proto.Call) {
 		}
 
 		// In next iteration, query next range.
-		args.Header().Key = descNext.StartKey
+		// Use the end key of current range rather than start key of next range,
```

Comment:
> // It's important that we use the `EndKey` of the current descriptor as opposed to the `StartKey` of the next one: If the former is stale, it's possible that the range has since merged with the subsequent one, and unless both are stale, the next descriptor's `StartKey` would equal the first's.